plugins/search-backend-module-pg/src/PgSearchEngine/PgSearchEngineIndexer.test.ts

```diff
@@ -156,4 +157,37 @@ describe('PgSearchEngineIndexer', () => {
     expect(result.error).toBe(expectedError);
     expect(tx.rollback).toHaveBeenCalledWith(expectedError);
   });
+
+  it('should rollback transaction on upstream error', async () => {
```

Comment:
Nice work getting a test in for this! Could maybe include additional tests for the cases where rollback should _not_ be attempted? (i.e. when there's no error, and when rollback has already happened after the addition of `isCompleted` below).

Reply:
Makes sense!  I'll add a `.not.toHaveBeenCalled()` to the "happy path" case and add another case for the `isCompleted` scenario.

Reply:
In addition to that happy path assertion, it ended up being simpler (if a little inelegant) to just add `expect(tx.rollback).toHaveBeenCalledTimes(1);` assertions to the existing error test cases.

Tests with the `TestPipeline` utility end up being more integration-y than unit-y, so I think it's fine.  I feel confident, at least!